src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlCommand.cs

```diff
@@ -4597,7 +4601,7 @@ private SqlDataReader RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavi
                     optionSettings = GetSetOptionsString(cmdBehavior);
                     if (returnStream)
                     {
-                        SqlClientEventSource.Log.TryTraceEvent("<sc.SqlCommand.ExecuteReader|INFO> {0}, Command executed as RPC.", ObjectID);
+                        SqlClientEventSource.Log.TryTraceEvent("SqlCommand.RunExecuteReaderTds | Info | Object Id {0}, Activity Id {1}, Client Connection Id {2}, SPID {3}, Command executed as RPC, RPC Name '{4}' ", ObjectID, ActivityCorrelator.Current, Connection?.ClientConnectionId, Connection?.ServerProcessId, rpc.rpcName);
```

Comment:
Will rpc be null here?

Reply:
Accessing `ActivityCorrelator.Current` can cause an allocation and pinvoke, can it be avoided unless the trace message is actually going to be printed?

Reply:
It's value is static, but the `ToString()` method getting called indeed concatenates string. I'll try to fetch Id and sequence instead of formatted string, but we need current activity Id in logs.

Reply:
> Will rpc be null here?

I'll add a check never-the-less.

Reply:
> It's value is static, 

It's thread static and lazy allocated which means the first touch on a thread will create it which is the allocation and pinvoke for guid that I referred to. It's not a huge cost but if it's just going to be ignored I'd rather not have the allocation happen.

In this case I can't see a good way to keep the capture semantic and avoid the allocation unless it's needed. I'll make a mental note to see if there's something I can do in future but don't worry about it here.